### DISCUSSION TITLE
Keep Zero Before Number

### DISCUSSION THREAD

**Discussion by jjcomg6** [JoshClose/CsvHelper#1868] [Q&A]
My Value Is 004,After Export To CSV File Change Number To 4, Delete Automatically Zero Before My Value How To Keep Zero Before Number?

**Comment by JoshClose** [comment]
What type is your value? If it's a number type, you can specify a formatter in your mapping.

```cs
void Main()
{
    var records = new List<Foo>
    {
        new Foo { Id = 1, Name = "one" },
    };
    var config = new CsvConfiguration(CultureInfo.InvariantCulture)
    {
    };
    using (var writer = new StringWriter())
    using (var csv = new CsvWriter(writer, config))
    {
        csv.Context.RegisterClassMap<FooMap>();
        csv.WriteRecords(records);
        writer.Dump();
    }
}

private class Foo
{
    public int Id { get; set; }
    public string Name { get; set; }
}

private class FooMap : ClassMap<Foo>
{
    public FooMap()
    {
        Map(m => m.Id).TypeConverterOption.Format("000");
        Map(m => m.Name);
    }
}
```
![image](https://user-images.githubusercontent.com/114971/134963593-17dd94a3-a2e7-463d-9b57-ac66ac153ee4.png)

**Reply by jjcomg6** [reply]
My Values Is String Fro Example 0001 But Export To CSV File Delete Zero Before My Number And Write 1,Your Solution Doesn't Work!

**Reply by vernetk** [reply]
Hi, 

I used the example of JoshClose, and by putting "0001" in the Name field which is a string then the export keeps "0".
Can you give an example? 

![image](https://user-images.githubusercontent.com/5453308/135820068-25d83d91-3b53-4c27-b483-2130e0d783e0.png)
